### DISCUSSION TITLE
I have a question regarding the color space conversion of the HDR function.

### DISCUSSION THREAD

**Discussion by xujunjie12345** [alvr-org/ALVR#2582] [General]
In the latest 21-dev version, after the HDR function is enabled, the linear space RGBA16F is adopted on both the server side and the client side.
On the server side, the linear values are actively converted into sRGB values for storage in FrameRender.fx.

```
if (correctionType == (uint)1) { // Left View sRGB
		color = LinearToNonlinearRGB(color, SRGB_GAMMA_TO_NONLINEAR);
	}
```

On the client side, the sRGB values are converted back into linear values in stream.wgsl.

```
    if ENABLE_SRGB_CORRECTION {
        let condition = vec3f(f32(color.r < THRESHOLD), f32(color.g < THRESHOLD), f32(color.b < THRESHOLD));
        let lowValues = color * DIV12;
        let highValues = pow((color + vec3f(0.055)) * DIV1, GAMMA);
        color = condition * lowValues + (1.0 - condition) * highValues;
    }
```

I wonder if these two conversions are redundant and waste performance.
I've tested that when these two conversions are blocked, there seems to be no change in the effect when HDR is turned on.
Is it because the precision loss of sRGB color values is smaller during GPU encoding?

**Comment by zmerp** [comment]
By specification, h264, HEVC and AV1 use the standard sRGB gamma curve (which is clamped near blacks). The conversion from sRGB to RGB is done because MediaCodec is not doing this conversion for whatever reason. Instead the Apple Vision Pro can use the decoded sRGB images without any extra conversion.

**Reply by xujunjie12345** [reply]
Thank you for your explanation.
Regarding what you mentioned that "h264, HEVC and AV1 use the standard sRGB gamma curve", is it referring to this?
```
            config.hevcVUIParameters.transferCharacteristics
                = NV_ENC_VUI_TRANSFER_CHARACTERISTIC_SRGB;
```

But I also see there are linear configurations:

`NV_ENC_VUI_TRANSFER_CHARACTERISTIC_LINEAR`

Similarly, AMD also has similar parameters:

`AMF_COLOR_TRANSFER_CHARACTERISTIC_LINEAR`

And from my tests, this parameter doesn't seem to have an obvious effect on the picture.
